src/test/java/org/example/pagefunctions/TimePage.java

```diff
@@ -0,0 +1,52 @@
+package org.example.pagefunctions;
+
+import org.example.commonfunctions.CommonFunctions;
+import org.example.objectrepo.OrangeHRM_ObjRepo;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.PageFactory;
+
+public class TimePage extends CommonFunctions {
+
+    OrangeHRM_ObjRepo objRepo;
+
+    public TimePage(WebDriver driver)
+    {
+        objRepo= PageFactory.initElements(driver,OrangeHRM_ObjRepo.class);
+    }
+
+    public void validateTimePage(WebDriver driver)
+    {
+        LoginPage loginPage=new LoginPage(driver);
+        loginPage.successfulLogin(driver);
+        waitForElementVisible(driver,objRepo.timeLink);
+        findButtonClick(objRepo.timeLink);
+        highlightElements(driver,objRepo.timeSheetsLinkTimePage);
+        findButtonClick(objRepo.timeSheetsLinkTimePage);
+        sleep(1);
+        highlightElements(driver,objRepo.myTimeSheetsLinkTimePage);
+        highlightElements(driver,objRepo.employeeTimeSheetsLink);
+        waitForElementVisible(driver,objRepo.attendanceLink);
+        highlightElements(driver,objRepo.attendanceLink);
+        findButtonClick(objRepo.attendanceLink);
+        sleep(1);
+        highlightElements(driver,objRepo.myRecordsLink);
+        highlightElements(driver,objRepo.punchInOutLink);
+        highlightElements(driver,objRepo.employeeRecordsLink);
+        highlightElements(driver,objRepo.configurationLinkOnTimePage);
+        highlightElements(driver,objRepo.reportsLinkTimePage);
+        findButtonClick(objRepo.reportsLinkTimePage);
+        sleep(1);
+        highlightElements(driver,objRepo.projectReportsLink);
+        highlightElements(driver,objRepo.employeeReportsLink);
+        highlightElements(driver,objRepo.attendanceSummaryLink);
+        highlightElements(driver,objRepo.projectInfoLink);
+        findButtonClick(objRepo.projectInfoLink);
```

Comment:
I am only seeing two or three classes, again it is because the rest of the code is already merged. So I am leaving the comment here. In this class I don't see any assertions, so I assume you used assertions in methods like findButtonClick, highlightElements in CommonFunctions class. If you do not have assertions in these methods, you have two options: 

add assertions to the methods in CommonFunctions class. Or add assertions in this particular class. either way you're validating.

Reply:
In some of the method I couldn't add assertion, because confirmation message, even the elements only displayed for a second and just disappeared, couldn't select the elements. I searched for a solution, but didn't find a way to solve it.

Reply:
Inspect the page, click "Sources" Tab, on the right side you'll see a pause icon, hover over it and you'll see something like "pause script execution F8 Command + \" , first you make sure what they are exactly in your machine. 

Second step is that right after you perform the action when the success message appears you press F8 or Command + \. So the application will be frozen right there with the success message. then you locate it.